peak_finder/driver.py

```diff
@@ -201,12 +201,15 @@ def run(self):  # pylint: disable=R0912, R0914, too-many-statements
                 masking_array = self.params.masking_data.values
 
                 workspace = Workspace()
-                survey = survey.copy(parent=workspace)
+                survey: Curve = survey.copy(parent=workspace)  # type: ignore
```

Comment:
**question:** why does it require to be ignored? would `cast` make sense and be better appropriate?

See https://docs.python.org/3/library/typing.html#typing.cast

Reply:
or is it because it could return null?

```suggestion
                survey = survey.copy(parent=workspace)
                assert survey is not None
```

Reply:
The copy method return type is any Entity, and survey is typed as Curve. We would need to change the return type to be same as input on geoh5py. I will cast for now.